### PR TITLE
update GO and operator versions

### PR DESCRIPTION
**Pull request by almoggue**
- update GO version to 1.21.5
- fix CVEs by updating GO packages
- update tests:
  - operator version 7.2.4-12
  - update k8s version 1.26
- enable running CI action manually

### CODE REVIEW

File: bootstrap/README.md [hunk 1]
```diff
@@ -54,7 +54,7 @@ kind: RedisEnterpriseDatabase
 metadata:
   name: mydb
 spec:
-  memory: 100MB
```
Comment (alon-zada) [inline bootstrap/README.md:57]: preserveUnknownFields: false

File: docs/guides/using-the-plugin-on-k8s.md [hunk 2]
```diff
@@ -72,7 +72,7 @@ kind: RedisEnterpriseDatabase
 metadata:
```
Comment (alon-zada) [inline docs/guides/using-the-plugin-on-k8s.md:72]: to be allied please consider adding the label to the REDB metadata here:
  labels:
    app: redis-enterprise

File: bootstrap/README.md [hunk 3]
```diff
@@ -54,7 +54,7 @@ kind: RedisEnterpriseDatabase
 metadata:
```
Comment (alon-zada) [inline bootstrap/README.md:54]: also here:
to be allied please consider adding the label to the REDB metadata here:
  labels:
    app: redis-enterprise

**Review by alon-zada — COMMENTED** [review]
Great!
Please view non-blocking comments.
like the declarative pipelines reminded me how easy they are 👍  although the testing imo should be by running a python script and not bash cmds in the yaml but unrelated to the pr.

**Review by alon-zada — APPROVED** [review]
see above comments :)

File: bootstrap/README.md (hunk 3 above)
Comment (almoggue) [reply bootstrap/README.md:54]: added

**Review by alon-zada — APPROVED** [review]
Great!

**Pull request merged**